src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java

```diff
@@ -449,37 +429,26 @@ public int compare(FeedbackSessionAttributes session1, FeedbackSessionAttributes
      * method is called with combined feedback sessions from many courses
      */
     public static void sortFeedbackSessionsByCreationTimeDescending(List<FeedbackSessionAttributes> sessions) {
-        Collections.sort(sessions, new Comparator<FeedbackSessionAttributes>() {
-            @Override
-            public int compare(FeedbackSessionAttributes session1, FeedbackSessionAttributes session2) {
-                int result = session2.createdTime.compareTo(session1.createdTime);
-                if (result == 0) {
-                    if (session1.endTime == null || session2.endTime == null) {
-                        if (session1.endTime == null) {
-                            --result;
-                        }
-                        if (session2.endTime == null) {
-                            ++result;
-                        }
-                    } else {
-                        result = session2.endTime.compareTo(session1.endTime);
-                    }
-                }
-
-                if (result == 0) {
-                    result = session2.startTime.compareTo(session1.startTime);
+        sessions.sort(Comparator.comparing((FeedbackSessionAttributes session) -> session.startTime).reversed()
+                .thenComparing(session -> session.courseId)
+                .thenComparing(session -> session.feedbackSessionName));
+        sessions.sort((session1, session2) -> {
+            int result = 0;
+
+            if (session1.endTime == null || session2.endTime == null) {
+                if (session1.endTime == null) {
+                    --result;
                 }
-                if (result == 0) {
-                    result = session1.courseId.compareTo(session2.courseId);
+                if (session2.endTime == null) {
+                    ++result;
                 }
-
-                if (result == 0) {
-                    result = session1.feedbackSessionName.compareTo(session2.feedbackSessionName);
-                }
-
-                return result;
+            } else {
+                result = session2.endTime.compareTo(session1.endTime);
             }
+
+            return result;
         });
+        sessions.sort(Comparator.comparing((FeedbackSessionAttributes session) -> session.createdTime).reversed());
```

Comment:
Forgive me but based on my understanding, for a stable sorting algorithm, if I intended to sort based on a chain of a-b-c, when I separate them in different parts, I need to reverse the order of sorting, meaning sorting based on c, then based on b, then based on a since the sorting is stable? Correct me if I am wrong.

Reply:
Therefore, if I just put `createdTime` into the chain, as `endTime` was previously in between `createdTime` and others, the order would be wrong. Or should I just put `createdTime` into the chain and for the `endTime`, sorting only if "session1.createdTime != session2.createdTime", else return 0?

Reply:
@LuLechuan coming back to what I asked before: what is the difficulty you faced when putting everything into one single chain? It is very much possible to do so.